packages/libs/error-stack/src/fmt/mod.rs

```diff
@@ -829,66 +677,101 @@ impl Opaque {
     }
 }
 
-fn debug_attachments(
-    loc: Option<Line>,
-    position: Position,
+fn debug_attachments_order(emits: impl Iterator<Item = Emit>) -> Vec<String> {
+    let mut out = vec![];
+    let mut defer = vec![];
+
+    for emit in emits {
+        match emit {
+            Emit::Defer(value) => {
+                defer.push(value);
+            }
+            Emit::Immediate(value) => {
+                out.push(value);
+            }
+        }
+    }
+
+    out.append(&mut defer);
```

Comment:
Shouldn't we use
```rust
out.extend(defer);
```
to consume `defer`?
(Maybe I'm missing something here, this is untested)

Reply:
We can, but `.append()` is more efficient.

> The difference between the functions is, that extend adds items one after another while, append adds all of them at once. This allows the destination data structure to decide how to add them efficiently. However, append is a bit less flexible than extend, as you always add the entire second data structure while the iterator of extend can be used to filter items.

from: https://github.com/rust-lang/rust-clippy/issues/4321

Reply:
I didn't know that `append` does move the elements. I think I never used that function :D